keras/backend/tensorflow_backend.py

```diff
@@ -3334,13 +3334,17 @@ def pool3d(x, pool_size, strides=(1, 1, 1), padding='valid',
     return _postprocess_conv3d_output(x, data_format)
 
 
-def bias_add(x, bias, data_format=None):
+def bias_add(x, bias, data_format=None, bias_shape=None):
```

Comment:
The purpose of this argument is to enhance the bias_add method to support add bias as a multi-dimension tensor. The existing bias_add only support to add a bias which is 1D tensor, but the bias we have in local_conv layer is a multi-dimension tensor (like (output_row, output_col, filters) for local-2d conv), so we need to know what is the target shape we expect the tensor to be before add to the output, that's why I add this argument. Otherwise we will have to put the handle of data_format and broadcast in the LocalConnected layer.

Or should we give it a better name?